src/control/cmd/daos_agent/security_rpc.go

```diff
@@ -48,46 +51,69 @@ type (
 	securityConfig struct {
 		credentials *security.CredentialConfig
 		transport   *security.TransportConfig
+		infoCache   *InfoCache
+		sys         string
 	}
 
 	// SecurityModule is the security drpc module struct
 	SecurityModule struct {
-		log            logging.Logger
-		signCredential credSignerFn
-		credCache      *credentialCache
+		log              logging.Logger
+		signCredential   credSignerFn
+		credCache        *credentialCache
+		config           *securityConfig
+		validAuthFlavors []auth.Flavor
+		infoCache        *InfoCache
+	}
 
-		config *securityConfig
+	authArgs struct {
+		tag     auth.AuthTag
+		reqBody []byte
 	}
+
+	AuthValidSet map[auth.AuthTag]bool
 )
 
 var _ cache.ExpirableItem = (*cachedCredential)(nil)
 
+func getAuthArgs(reqb []byte) (*auth.AuthArgs, error) {
```

Comment:
The caller should pass the flavor I am using, I can change the client logic to ping the agent and ask which flavor to use.